todos/src/actions/index.js

```diff
@@ -1,33 +1,85 @@
-export const addTodo = (data) => {
-  return {
-    type: "ADD_TODO",
-    payload: {
-      id: new Date().getTime().toString(),
+import {
+  addTodoSupabase,
+  deleteTodoSupabase,
+  updateStateTodoSupabase,
+  updateStateTodoSupabaseOnEdit,
+  updateTodoSupabase,
+} from "supabaseData";
+import {
+  STATE_CHANGE_MESSAGE,
+  EDIT_SUCCESS_MESSAGE,
+  TASK_ADDED_MESSAGE,
+  TASK_DELETE_MESSAGE,
+  WENT_WRONG_MESSAGE,
+} from "constants";
+import { successMessage, dangerMessage } from "toastMethods";
+
+export const addTodo = (data) => async (dispatch) => {
+  try {
+    dispatch(setAddCardLoadingState(true));
+    const response = await addTodoSupabase(
       data,
-      isCompleted: false,
-      date: new Date().toLocaleDateString(),
-      completedDate: null,
-      dateNow: Date.now(),
-      onEdit: false,
-    },
-  };
+      new Date().toLocaleDateString()
```

Comment:
Can we pass date from the component that's dispatching `addTodo`? Actions should be deterministic in nature

Reply:
Yes, Bhai. This one done and also changed for the others.